OpenRA.Mods.Common/Traits/World/ActorSpawnManager.cs

```diff
@@ -77,7 +77,10 @@ void ITick.Tick(Actor self)
 			if (spawnPoint == null)
 				return;
 
-			spawnCountdown = info.SpawnInterval;
+			if (info.SpawnInterval.Length == 1)
+				spawnCountdown = info.SpawnInterval[0];
+			else if (info.SpawnInterval.Length == 2)
+				spawnCountdown = self.World.SharedRandom.Next(info.SpawnInterval[0], info.SpawnInterval[1]);
```

Comment:
Shouldn't this use `Util.RandomDelay`? I'm also not sure about explicitly requiring a length of two. If we want to do this properly we should throw in `IRulesetLoaded` when `SpawnInterval` has 0 or more than 2 entries.

Reply:
> Shouldn't this use `Util.RandomDelay`?

Yep, good call.

> If we want to do this properly we should throw in IRulesetLoaded when SpawnInterval has 0 or more than 2 entries.

Will do.